Mage.Sets/src/mage/cards/b/BragoKingEternal.java

```diff
@@ -34,14 +40,11 @@ public BragoKingEternal(UUID ownerId, CardSetInfo setInfo) {
 
         // Flying
         this.addAbility(FlyingAbility.getInstance());
+
         // When Brago, King Eternal deals combat damage to a player, exile any number of target nonland permanents you control, then return those cards to the battlefield under their owner's control.
-        Effect effect = new ExileTargetEffect(this.getId(), this.getName(), Zone.BATTLEFIELD);
-        effect.setText("exile any number of target nonland permanents you control");
-        Ability ability = new DealsCombatDamageToAPlayerTriggeredAbility(effect, false);
-        FilterControlledPermanent filterControlledNonlandPermanent = new FilterControlledPermanent();
-        filterControlledNonlandPermanent.add(Predicates.not(CardType.LAND.getPredicate()));
-        ability.addTarget(new TargetControlledPermanent(0, Integer.MAX_VALUE, filterControlledNonlandPermanent, false));
-        ability.addEffect(new ReturnFromExileEffect(this.getId(), Zone.BATTLEFIELD, ", then return those cards to the battlefield under their owner's control"));
+        Ability ability = new DealsCombatDamageToAPlayerTriggeredAbility(new ExileTargetForSourceEffect().setText("exile any number of target nonland permanents you control"), false);
```

Comment:
`ExileTargetForSourceEffect` must be used with `ReturnFromExileForSourceEffect`

Reply:
It should work either way, shouldn't it?

Reply:
If it uses same source and zcc related zones then all fine. Old version used different zones.